### ISSUE TITLE
rumdl is parsing fenced codeblocks that are tagged with `markdown` as the language

### ISSUE THREAD

**Issue by kattni** [rvben/rumdl#380]
### rumdl version

0.1.9

### Installation method

Other

### Bug description

rumdl is parsing the contents of fenced codeblocks, when the code language is set to `markdown`. 

### Steps to reproduce

1. Create a test file with the following content:

```markdown
[Link text]()
```

2. Run rumdl.

### Expected behavior

rumdl should ignore the contents of any fenced codeblock. I should be able to put invalid Markdown inside a fenced code block for demonstration purposes, and have it properly highlighted.

### Actual behavior

rumdl fails on the demo code above with `[MD042] Empty link found: [Link text]()`. 

I actually found this failing on trying to parse an invalid anchor name because it's meant to be a fake example of what a Markdown anchor link looks like.

### Configuration

```toml
[tool.rumdl]
flavor = "mkdocs"
include = ["**/*.md"]
exclude = ["docs/en/SUMMARY.md", "comment.md"]
respect-gitignore = true

# Disable rules:
# MD014: Force commands in codeblocks to show output
disable = ["MD014"]

[tool.rumdl.per-file-ignores]
# MD002: First heading should be level 1, found level 2; title will be included in the target page
# MD034: URL without angle brackets or link formatting; triggers on URLs with Jinja variable included
# MD041: First line in file should be level 1 heading; title will be included in the target page
"src/beeware_docs_tools/shared_content/en/*.md" = ["MD002", "MD034", "MD041",]

[tool.rumdl.MD013]  # Line length
line_length = 999999  # Force reflow to paragraph content to remove linebreaks
reflow = true
reflow_mode = "normalize"

[tool.rumdl.MD026]  # Remove punctuation at the end of headings
punctuation = ",;:"

[tool.rumdl.MD033]  # No inline HTML
allowed_elements = ["nospell",]  # pyspelling inline disable tag

[tool.rumdl.MD061]
terms = ["TODO"]
```

### Operating system

macOS

**Comment by rvben** [comment]
Hi @kattni, this is actually a feature. But.. you are probably right that this should not be enabled by default and should be handled by the new codeblock tools logic. So this will be resolved in the upcoming release.

**Comment by rvben** [comment]
Fixed in v0.1.13. Embedded markdown linting is now opt-in - it requires explicit configuration:

```toml
[code-block-tools.languages.md]
lint = ["rumdl"]
```

Without this configuration, markdown code blocks are no longer linted by default.
